flink-formats/flink-orc/src/main/java/org/apache/flink/orc/vector/AbstractOrcColumnVector.java

```diff
@@ -0,0 +1,198 @@
+
+/**
+ * This column vector is used to adapt hive's ColumnVector to Flink's ColumnVector.
+ */
+public abstract class AbstractOrcColumnVector implements
+		org.apache.flink.table.dataformat.vector.ColumnVector {
+
+	private ColumnVector vector;
+
+	AbstractOrcColumnVector(ColumnVector vector) {
+		this.vector = vector;
+	}
+
+	@Override
+	public boolean isNullAt(int i) {
+		return !vector.noNulls && vector.isNull[vector.isRepeating ? 0 : i];
+	}
+
+	@Override
+	public void reset() {
+		throw new UnsupportedOperationException();
+	}
+
+	public static org.apache.flink.table.dataformat.vector.ColumnVector createVector(
+			ColumnVector vector) {
+		if (vector instanceof LongColumnVector) {
+			return new OrcLongColumnVector((LongColumnVector) vector);
+		} else if (vector instanceof DoubleColumnVector) {
+			return new OrcDoubleColumnVector((DoubleColumnVector) vector);
+		} else if (vector instanceof BytesColumnVector) {
+			return new OrcBytesColumnVector((BytesColumnVector) vector);
+		} else if (vector instanceof DecimalColumnVector) {
+			return new OrcDecimalColumnVector((DecimalColumnVector) vector);
+		} else if (vector instanceof TimestampColumnVector) {
+			return new OrcTimestampColumnVector((TimestampColumnVector) vector);
+		} else {
+			throw new UnsupportedOperationException("Unsupport vector: " + vector.getClass().getName());
+		}
+	}
+
+	/**
+	 * Create flink vector by hive vector from constant.
+	 */
+	public static org.apache.flink.table.dataformat.vector.ColumnVector createVectorFromConstant(
+			LogicalType type, Object value, int batchSize) {
+		return createVector(createHiveVectorFromConstant(type, value, batchSize));
+	}
+
+	/**
+	 * Create a orc vector from partition spec value.
+	 * See hive {@code VectorizedRowBatchCtx#addPartitionColsToBatch}.
+	 */
+	private static ColumnVector createHiveVectorFromConstant(
+			LogicalType type, Object value, int batchSize) {
+		switch (type.getTypeRoot()) {
+			case CHAR:
+			case VARCHAR:
+			case BINARY:
+			case VARBINARY:
+				return createBytesVector(batchSize, value);
+			case BOOLEAN:
+				return createLongVector(batchSize, (Boolean) value ? 1 : 0);
+			case TINYINT:
+			case SMALLINT:
+			case INTEGER:
+			case BIGINT:
+				return createLongVector(batchSize, value);
+			case DECIMAL:
+				DecimalType decimalType = (DecimalType) type;
+				return createDecimalVector(
+						batchSize, decimalType.getPrecision(), decimalType.getScale(), value);
+			case FLOAT:
+			case DOUBLE:
+				return createDoubleVector(batchSize, value);
+			case DATE:
+				return createLongVector(batchSize, dateToInternal((Date) value));
+			case TIMESTAMP_WITHOUT_TIME_ZONE:
+				return createTimestampVector(batchSize, (LocalDateTime) value);
+			default:
+				throw new UnsupportedOperationException("Unsupported type: " + type);
+		}
+	}
+
+	private static LongColumnVector createLongVector(int batchSize, Object value) {
+		LongColumnVector lcv = new LongColumnVector(batchSize);
```

Comment:
I think it can be kept as it is now, because it can keep all vectors consistent and support updates more flexibly.

Reply:
why do you want to update them?

Reply:
I don't want to update them.But if we want to update them in the future.
I think we expose a consistent `VectorizedColumnBatch` is good to a clean semantics.

Reply:
mark the vector repeating didn't break anything, right? For the normal vector you read, it might also be repeated, and you already handled it during read. 
Using a repeated vector is good for memory.

Reply:
- It breaks the design of orc `ColumnVector`,  orc `ColumnVector` provides `fill` and `setElement` to update vector, if we set a size 1, There is no doubt that it will destroy this behavior.
- I don't think we should make these vectors special. Regular vectors always have correct size.
- About good for memory, I don't think there's so much memory. And it has no impact on performance.

Reply:
I read the source code, and actually `fill` only set the first element and will also set `isRepeated` to `true`. In this case, other fields will be `NULL`, sounds acceptable to me.